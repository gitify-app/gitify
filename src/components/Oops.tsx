import { type FC, useMemo } from 'react';

import type { GitifyError } from '../types';

interface IProps {
  error: GitifyError;
}

export const Oops: FC<IProps> = ({ error }) => {
  const emoji = useMemo(
    () => error.emojis[Math.floor(Math.random() * error.emojis.length)],
    [error],
  );

  return (
    <div className="flex flex-1 flex-col justify-center items-center p-4 bg-white dark:bg-gray-dark text-black dark:text-white">
      <h1 className="text-5xl mb-5">{emoji}</h1>

      <h2 className="font-semibold text-xl mb-2 text-semibold">
        {error.title}
      </h2>
      {error.descriptions.map((description, i) => {
        return (
          // biome-ignore lint/suspicious/noArrayIndexKey: using index for key to keep the error constants clean
          <div className="text-center mb-2" key={`error_description_${i}`}>
            {description}
          </div>
        );
      })}
    </div>
  );
};
