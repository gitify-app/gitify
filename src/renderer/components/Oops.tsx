import { type FC, useMemo } from 'react';

import { Stack } from '@primer/react';
import type { GitifyError } from '../types';
import { Centered } from './primitives/Centered';
import { EmojiText } from './primitives/EmojiText';

interface IOops {
  error: GitifyError;
}

export const Oops: FC<IOops> = ({ error }: IOops) => {
  const emoji = useMemo(
    () => error.emojis[Math.floor(Math.random() * error.emojis.length)],
    [error],
  );

  return (
    <Centered>
      <Stack direction="vertical" align="center">
        <div className="mt-2 text-5xl">
          <EmojiText text={emoji} />
        </div>

        <div className="mb-2 text-xl font-semibold">{error.title}</div>
        {error.descriptions.map((description, i) => {
          return (
            // biome-ignore lint/suspicious/noArrayIndexKey: using index for key to keep the error constants clean
            <div className="mb-2 text-center" key={`error_description_${i}`}>
              {description}
            </div>
          );
        })}
      </Stack>
    </Centered>
  );
};
