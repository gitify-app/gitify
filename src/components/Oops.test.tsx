import * as TestRenderer from 'react-test-renderer';

import { Oops } from './Oops';

describe('components/Oops.tsx', () => {
  it('should render itself & its children', () => {
    const mockError = {
      title: 'Error title',
      descriptions: ['Error description'],
      emojis: ['ðŸ”¥'],
    };
    const tree = TestRenderer.create(<Oops error={mockError} />);

    expect(tree).toMatchSnapshot();
  });
});
