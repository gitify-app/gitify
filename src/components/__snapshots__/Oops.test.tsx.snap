// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Oops.tsx should render itself & its children 1`] = `
<div
  className="flex flex-1 flex-col justify-center items-center p-4 bg-white dark:bg-gray-dark text-black dark:text-white"
>
  <h1
    className="text-5xl mb-5"
  >
    ðŸ”¥
  </h1>
  <h2
    className="font-semibold text-xl mb-2 text-semibold"
  >
    Error title
  </h2>
  <div
    className="text-center mb-2"
  >
    Error description
  </div>
</div>
`;
